snapcraft/remote/utils.py

```diff
@@ -76,3 +78,27 @@ def _compute_hash(directory: Path) -> str:
 
     all_hashes = "".join(hashes).encode()
     return md5(all_hashes).hexdigest()  # noqa: S324 (insecure-hash-function)
+
+
+def rmtree(directory: Path) -> None:
+    """Cross-platform rmtree implementation.
+
+    :param directory: Directory to remove.
+    """
+    shutil.rmtree(
+        str(directory.resolve()),
+        onerror=_remove_readonly,  # type: ignore
+    )
+
+
+def _remove_readonly(func, filepath):
+    """Shutil onerror function to make read-only files writable.
+
+    Try setting file to writeable if error occurs during rmtree. Known to be required
+    on Windows where file is not writeable, but it is owned by the user (who can
+    set file permissions).
+
+    :param filepath: filepath to make writable
+    """
+    filepath.chmod(stat.S_IWRITE)
+    func(filepath)
```

Comment:
What happens if this fails again?

Reply:
It depends on what function is passed, but the I believe it will raise an OS or shutil error.